modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java

```diff
@@ -184,6 +185,10 @@ private FileEntry _addTempImageFileEntry(PortletRequest portletRequest)
 
 		File file = uploadPortletRequest.getFile("fileName");
 
+		if (file.length() > maxFileSize) {
```

Comment:
Not super familiar with this module, but looking at _getMaxSize(), it seems like the parameter maxFilzeSize could be passed in by user. I wonder if this condition is enough to prevent the intended purpose?

Reply:
No it cannot, maxFileSize is defined here: https://github.com/liferay-appsec/liferay-portal/blob/76796ea576bd26246db88fa5b2319d321cb81b0a/modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java#L118
And in case its a user profile picture, we are getting it from the configuration, not from the frontend

Reply:
@NemethNorbert 
I understand the local variable of maxFileSize is from the line 118, but I was talking about the implementation of getMaxFileSize(), specifically:
https://github.com/liferay-appsec/liferay-portal/blob/76796ea576bd26246db88fa5b2319d321cb81b0a/modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java#L243

Here you can see, in certain scenerios, it seems the value comes from request parameter.

Reply:
@arthurchan35 
I am not familiar with this module either, but the if statement above makes sure we don't get it from the request in case its a user profile picture of any kind.

Reply:
@arthurchan35 
I'll try to upload some pictures in the portal and see if end up there

Reply:
@NemethNorbert 
I agree that that if condition makes sure cases like user profile will have maxFileSize from configuration, but I was mostly concerned with other cases that (potentially) user inputted maxFileSize.

@stian-sigvartsen 
Do you have opinion on this topic?
It seems that you and Tibor were discussing if if core-infra team is more appropreate to review this PR, should it be forwarded to them?

Reply:
@NemethNorbert 
I think we can definitely have this PR forwarded to fix the current specific problem with profile picture.

But maybe leave a question to the author's team will help raise the concern.

Reply:
Sounds like a good plan, if anyone the responsible team for this module should know where it could be a concern

Reply:
In terms of fixing the current specific problem, I can see the fix relies on another request parameter: "currentLogoURL". That could also be manipulated to circumvent the fix, or do we have other protective measures validating that?

Reply:
@stian-sigvartsen,

You are right, modifying currentLogoURL would bypass the validation I put there. I think the solution is to get the maxFileSize from the configuration either way, but I'd assume this size is different for simple document uploads.
I have to dig in here a bit. Thanks for pointing that out

Reply:
This private method was introduced in https://github.com/liferay-appsec/liferay-portal/commit/3410ea72b8daa519af17e10c89a2ddf201ab9773
And I think it was only written like this to make sure it does not broke anything. I decided to remove it altogether, as I tested all the modules including commerce with image and none of them using this module.
I think we should just make sure to let the responsible team know about this deletion and they can point out why this logical check was necessary if at all.

Thanks,